docs/storage/prometheus.md

```diff
@@ -78,6 +78,7 @@ Metric name | Type | Description | Unit (where applicable)
 `container_spec_memory_reservation_limit_bytes` | Gauge | Memory reservation limit for the container | bytes
 `container_start_time_seconds` | Gauge | Start time of the container since unix epoch | seconds
 `container_tasks_state` | Gauge | Number of tasks in given state (`sleeping`, `running`, `stopped`, `uninterruptible`, or `ioawaiting`) |
+`container_wss` | Gauge |  Container referenced bytes during last measurements cycle based on Referenced field in /proc/smaps file, with /proc/PIDs/clear_refs set to 1 after defined number of cycles configured through `wss_reset_interval` cAdvisor parameter.</br>Warning: this is intrusive collection, because can influence kernel page reclaim policy and add latency. Refer to https://github.com/brendangregg/wss#wsspl-referenced-page-flag for more details. | bytes
```

Comment:
Spitballing here, but can't we still rely on cgroup memory accounting to be correct?  Would it be enough just to use proc/PIDs/clear_refs to force reclaim?

The way to check would be to see if our memory metrics (e.g. usage_bytes, working_set or rss) match what we get from the new container_wss metric when it is turned on.

Reply:
Interval to reset "referenced flags" should be optional and orthogonal to "container_referenced_bytes" collection, the "referenced_bytes" itself may be useful for application analysis (referenced bytes are mapped to process pages with accessed/young bits) and are also cleared by kswapd during periodic scan or low memory conditions, but...

You're right about that "memcg concept" and big thanks for pointing that (but the formula should also include inactive_anon_page). memcg approach is based on pages belonging to different LRU list (active/inactive) rather than "accessed bit" raised by HW . My gut felling is that, "referenced approach" is more accurate - but cannot dispute more without empirical results - let us perform more detailed analysis comparing both approaches before proceeding.

Reply:
Looking at the man page for clear_refs, I can't actually tell if it is actually inducing reclaim or primarily updating accounting references.  My suggestion is probably only reasonable if it is doing the former...

Reply:
@ppalucki I know that you've made some experiments, please share you observations ;)

Reply:
That works for me.  Thanks for checking